universe/ValueRefs.h

```diff
@@ -1114,7 +1079,31 @@ ComplexVariable<T>::ComplexVariable(S&& variable_name,
     m_int_ref3(std::move(int_ref3)),
     m_string_ref1(std::move(string_ref1)),
     m_string_ref2(std::move(string_ref2))
-{}
```

Comment:
Can we merge those constructors somehow?

Reply:
With the `const char* variable_name` overload or to avoid the perfect forwarding template?

It could take a `string` by value and always move but as far as I understand that's not optimal if passed an lvalue.

Reply:
The main gripe with this is the duplicated code for calculating the invariance, not the parameters.

Reply:
Moved into an init function and added a common `ValueRefBase` struct to facilitate ranged processing.

(`Condition` could also share the same base class which would similarly clean up some condition invariant init code.)